src/circuit/gadget/utilities/cond_swap.rs

```diff
@@ -0,0 +1,319 @@
+        dev::MockProver,
+        plonk::{Any, Assignment, Circuit, Column, ConstraintSystem, Error},
+    };
+    use pasta_curves::{arithmetic::FieldExt, pallas::Base};
+
+    #[test]
+    fn cond_swap() {
+        struct MyCircuit<F: FieldExt> {
+            x: Option<F>,
+            y: Option<F>,
+            swap: Option<bool>,
+        }
+
+        impl<F: FieldExt> Circuit<F> for MyCircuit<F> {
+            type Config = CondSwapConfig;
+
+            fn configure(meta: &mut ConstraintSystem<F>) -> Self::Config {
+                let advices = [
+                    meta.advice_column(),
+                    meta.advice_column(),
+                    meta.advice_column(),
+                    meta.advice_column(),
+                    meta.advice_column(),
+                ];
+
+                let perm = meta.permutation(
+                    &advices
+                        .iter()
+                        .map(|advice| (*advice).into())
+                        .collect::<Vec<Column<Any>>>(),
+                );
+
+                CondSwapChip::<F>::configure(meta, advices, perm)
+            }
+
+            fn synthesize(
+                &self,
+                cs: &mut impl Assignment<F>,
+                config: Self::Config,
+            ) -> Result<(), Error> {
+                let mut layouter = SingleChipLayouter::new(cs)?;
+                let chip = CondSwapChip::<F>::construct(config.clone());
+
+                // Load the pair and the swap flag into the circuit.
+                let x = chip.load_private(layouter.namespace(|| "x"), config.x, self.x)?;
+                let y = chip.load_private(layouter.namespace(|| "y"), config.y, self.y)?;
+                // Return the swapped pair.
+                let swapped_pair =
+                    chip.swap(layouter.namespace(|| "swap"), (x, y).into(), self.swap)?;
+
+                if let Some(swap) = self.swap {
+                    if swap {
+                        // Check that `x` and `y` have been swapped
+                        assert_eq!(swapped_pair.0.value.unwrap(), y.value.unwrap());
+                        assert_eq!(swapped_pair.1.value.unwrap(), x.value.unwrap());
+                    } else {
+                        // Check that `x` and `y` have not been swapped
+                        assert_eq!(swapped_pair.0.value.unwrap(), x.value.unwrap());
+                        assert_eq!(swapped_pair.1.value.unwrap(), y.value.unwrap());
+                    }
+                }
+
+                Ok(())
+            }
+        }
+
+        // Test swap case
+        {
+            let circuit: MyCircuit<Base> = MyCircuit {
+                x: Some(Base::rand()),
+                y: Some(Base::rand()),
+                swap: Some(true),
+            };
+            let prover = match MockProver::<Base>::run(3, &circuit, vec![]) {
+                Ok(prover) => prover,
+                Err(e) => panic!("{:?}", e),
+            };
```

Comment:
This can simply `unwrap`, which achieves the same result (panicking with the debug format of `e` on failure):
```suggestion
            let prover = MockProver::<Base>::run(3, &circuit, vec![]).unwrap();
```